### PR TITLE
[NFC] Upgrade EnumElementDecl to a DeclContext

### PR DESCRIPTION
**Pull request by CodaFi**
Pure plumbing for the sake of default arguments.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Review by slavapestov — APPROVED** [review]

**Comment by slavapestov** [comment]
@DougGregor @jckarter Can you take a quick look too? It's easy to miss a spot when adding a new kind of thing like this.

**Comment by jrose-apple** [comment]
Oh, good, this'll make it easier for @griotspeak to add GADTs to Swift 15.

File: lib/Serialization/Serialization.cpp [hunk 1]
```diff
@@ -1864,6 +1866,7 @@ void Serializer::writeCrossReference(const DeclContext *DC, uint32_t pathLen) {
   case DeclContextKind::Initializer:
   case DeclContextKind::TopLevelCodeDecl:
   case DeclContextKind::SerializedLocal:
+  case DeclContextKind::EnumElementDecl:
     llvm_unreachable("cannot cross-reference this context");
```
Comment (jrose-apple) [inline lib/Serialization/Serialization.cpp]: If EnumElementDecls grow proper ParamDecls, I think you *will* need to cross-reference this context.

**Comment by jckarter** [comment]
Like Jordan said, it seems like a good idea to set up xrefs for these now so you won't run into problems with them missing in the future.

**Review by DougGregor — COMMENTED** [review]
I don't see any issues here beyond what's already been noted. Asserting that we don't cross-reference these as contexts seems fine for me... adding untestable xref plumbing wouldn't really buy us anything.

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test Linux platform

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
In the interest of getting this whole thing moving again, I'm going to leave the XREF plumbing out and merge.

**Pull request merged**